t/basic.t

```diff
@@ -2481,7 +2499,14 @@ subtest "exeptions are not thrown twice" => sub {
 
   $app->processor->clear_exceptions;
 
-  my (undef, $res) = capture_stderr(sub { $jmap_tester->ua->get($uri) });
+  my (undef, $res) = capture_stderr(sub {
+    $jmap_tester->ua->request(
+      $jmap_tester,
+      HTTP::Request->new(GET => $uri),
+      'jmap',
+    )->get;
```

Comment:
Hm. This `->get` also feels like a smell to me. That's because: in general, if you just have a JMAP::Tester, you don't need to _know_ that it traffics in Futures internally, and only if you're doing stuff with its UA directly.

It feels to me like the UA should inherit the Tester's `should_return_futures` setting (which would be possible if there were, as suggested elsewhere, a weak ref back to the tester from the UA). Then `$tester->ua->request` would just do the right thing, based on how you'd configured the tester, and you wouldn't have this telltale `->get`. (I know that it isn't _really_ an implementation detail, because `->request` is doecumented to return a Future, but it still _feels_ like one, IMO.)

Reply:
This has been resolved with the http_X methods.